process/rating/ratingsData.go

```diff
@@ -409,7 +459,65 @@ func (rd *RatingsData) ShardChainRatingsStepHandler() process.RatingsStepHandler
 }
 
+// SetStatusHandler sets the provided status handler if not nil
+func (rd *RatingsData) SetStatusHandler(handler core.AppStatusHandler) error {
+	if check.IfNil(handler) {
+		return process.ErrNilAppStatusHandler
+	}
+
+	rd.mutStatusHandler.Lock()
+	rd.statusHandler = handler
+	rd.mutStatusHandler.Unlock()
+
+	return nil
+}
+
 // IsInterfaceNil returns true if underlying object is nil
 func (rd *RatingsData) IsInterfaceNil() bool {
 	return rd == nil
 }
+
+func (rd *RatingsData) updateRatingsMetrics(epoch uint32) {
+	rd.mutStatusHandler.RLock()
+	defer rd.mutStatusHandler.RUnlock()
+
+	currentShardRatingsStep, _ := getRatingStepsForEpoch(epoch, rd.ratingsSetup.ShardChain.RatingStepsByEpoch)
+	rd.statusHandler.SetUInt64Value(common.MetricRatingsShardChainHoursToMaxRatingFromStartRating, uint64(currentShardRatingsStep.HoursToMaxRatingFromStartRating))
+	rd.statusHandler.SetStringValue(common.MetricRatingsShardChainProposerValidatorImportance, fmt.Sprintf("%f", currentShardRatingsStep.ProposerValidatorImportance))
+	rd.statusHandler.SetStringValue(common.MetricRatingsShardChainProposerDecreaseFactor, fmt.Sprintf("%f", currentShardRatingsStep.ProposerDecreaseFactor))
+	rd.statusHandler.SetStringValue(common.MetricRatingsShardChainValidatorDecreaseFactor, fmt.Sprintf("%f", currentShardRatingsStep.ValidatorDecreaseFactor))
+	rd.statusHandler.SetStringValue(common.MetricRatingsShardChainConsecutiveMissedBlocksPenalty, fmt.Sprintf("%f", currentShardRatingsStep.ConsecutiveMissedBlocksPenalty))
+
+	currentMetaRatingsStep, _ := getRatingStepsForEpoch(epoch, rd.ratingsSetup.MetaChain.RatingStepsByEpoch)
+	rd.statusHandler.SetUInt64Value(common.MetricRatingsMetaChainHoursToMaxRatingFromStartRating, uint64(currentMetaRatingsStep.HoursToMaxRatingFromStartRating))
+	rd.statusHandler.SetStringValue(common.MetricRatingsMetaChainProposerValidatorImportance, fmt.Sprintf("%f", currentMetaRatingsStep.ProposerValidatorImportance))
+	rd.statusHandler.SetStringValue(common.MetricRatingsMetaChainProposerDecreaseFactor, fmt.Sprintf("%f", currentMetaRatingsStep.ProposerDecreaseFactor))
+	rd.statusHandler.SetStringValue(common.MetricRatingsMetaChainValidatorDecreaseFactor, fmt.Sprintf("%f", currentMetaRatingsStep.ValidatorDecreaseFactor))
+	rd.statusHandler.SetStringValue(common.MetricRatingsMetaChainConsecutiveMissedBlocksPenalty, fmt.Sprintf("%f", currentMetaRatingsStep.ConsecutiveMissedBlocksPenalty))
+}
+
+func getRatingStepsForEpoch(epoch uint32, ratingStepsPerEpoch []config.RatingSteps) (config.RatingSteps, bool) {
+	var ratingSteps config.RatingSteps
+	found := false
+	for _, ratingStepsForEpoch := range ratingStepsPerEpoch {
+		if ratingStepsForEpoch.EnableEpoch <= epoch {
+			ratingSteps = ratingStepsForEpoch
+			found = true
+		}
+	}
+
+	return ratingSteps, found
+}
+
+func getChainParamsForEpoch(epoch uint32, chainParamsList []config.ChainParametersByEpochConfig) (config.ChainParametersByEpochConfig, bool) {
```

Comment:
Seems all right, assuming `ratingStepsPerEpoch` and `chainParamsList` are nicely sorted arrays (by enabled epoch). Is assumption good?

Reply:
added slices.Sort first, just to avoid any possible config issue